packages/tools/src/stateManagement/segmentation/config/segmentationColor.ts

```diff
@@ -98,13 +98,15 @@ function getSegmentIndexColor(
 }
 
 function setSegmentIndexColor(
-  segmentationRepresentationUID: string,
+  viewportId: string,
+  segmentationId: string,
   segmentIndex: number,
   color: Types.Color
 ): void {
   // Get the reference to the color in the colorLUT.
   const colorReference = getSegmentIndexColor(
-    segmentationRepresentationUID,
+    viewportId,
+    segmentationId,
     segmentIndex
   );
 
```

Comment:
Since the color values are directly modified, when we add color luts, do we ensure we take a full copy so we don't later on modify it?

Reply:
Yes during add we do